test/yargs-parser.js

```diff
@@ -762,6 +762,29 @@ describe('yargs-parser', function () {
       argv.should.have.property('n', 123)
     })
 
+    it('should set n to the numeric value 123, with n at the end of a group', function () {
+      var argv = parser([ '-ab5n123' ])
```

Comment:
I would honestly expect it to be parsed like @bcoe noted in the first comment. I've actually not seen it in other bins, but I would expect grouping short options to not offer assigning values at all, because it's not really obvious to which options the values shall apply. As an example:
`-xy1` could mean:
1) Assign `1` to `x` and `y`, or
2) Assign `1` to `y` and `true` to `x`

Even if we would document it, it's just not obvious if you read something like this.
Additionally, currently we do assigning values to options with either a `=` or a space:
`--value=1` and `--value 1`
if we now allowed typing values _of a certain kind_  right after the name of the option, wouldn't that kinda break the consistency?